core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -1094,14 +1094,14 @@ private void checkPredicates(RelOptCluster cluster, RelOptTable empTable,
   }
 
   @Test public void testPullUpPredicatesFromProject() {
-    final String sql = "select deptno, mgr, x, 'y' as y from (\n"
-        + "  select deptno, mgr, cast(null as integer) as x\n"
+    final String sql = "select deptno, mgr, x, 'y' as y, z from (\n"
+        + "  select deptno, mgr, cast(null as integer) as x, cast('1' as int) as z\n"
         + "  from emp\n"
         + "  where mgr is null and deptno < 10)";
     final RelNode rel = convertSql(sql);
     RelOptPredicateList list = RelMetadataQuery.getPulledUpPredicates(rel);
     assertThat(list.pulledUpPredicates.toString(),
-        is("[IS NULL($1), <($0, 10), IS NULL($2), =($3, 'y')]"));
+        is("[IS NULL($1), <($0, 10), IS NULL($2), =($4, CAST('1'):INTEGER NOT NULL), =($3, 'y')]"));
```

Comment:
Is this "toString" representation sorted?
If not, it could be subject to different order depending on JRE version/implementation